### DISCUSSION TITLE
Beatstep encoders

### DISCUSSION THREAD

**Discussion by adamgd** [raphaelquast/beatstep#75] [Q&A]
I recently got a Beatstep with the intention to use Beatstep_Q. The scripts are great overall!

That said, I'm frustrated at the lack of response from the encoders. I seem to be having the problem described [here](https://forum.arturia.com/index.php?topic=82122.0) where it takes multiple rotations to traverse the range of CC values regardless of acceleration values in MCC. I realise this is probably not a Beatstep_Q problem per se, but I'm wondering if it is universal and those using Beatstep_Q cope with it or find ways to work around it, or if I'm unlucky or doing something wrong.

Any advice?

**Comment by raphaelquast** [comment]
Hey, sorry for the late response!

I too am experiencing problems with some of the encoders... but my BeatStep is several years old... so some problems might come from wear and tear over the years (they are jumpy and sometimes do not respond correctly to changes in the rotation-direction)

So far I have tried several workarounds (some of them implemented in the script) that try to deal with some of the encoder-problems, 
but I have not yet found a solution that make all of the encoders behave nicely. (as I said, on my side, some do, some dont...)
As far as I know this is a problem that can only be solved 100% by Arturia itself (most probably by properly updating the firmware... their technical support was helpful when I had some questions, but I doubt that they will release an updated firmware soon)

This was also one of the reasons why I implemented the "multi-touch-editing" in the `sequence-editor`
(it uses the transpose-encoder (which is working fine on my side) to do all the changes and avoid using the buggy encoders...)


If you can explain your problems in more detail I can check if I'm experiencing the same issues... if so we might think of ways to circumvent the encoder-limitations that can be implemented in the script.

**Reply by adamgd** [reply]
@raphaelquast Thanks for your response. I purchased the Beatstep recently and kept it as long as I could within the product return window but sent it back earlier this week so my ability to help work through this issue is limited.

I'm mainly interested in DJing and remixing with a controller, and so I want to mimic the quick, single gesture (<1 full rotation) action of turning a pot (eg EQ controls) on a DJ mixer. I experienced the same problem with all encoders on a brand new unit: it takes too many turns to cover the full domain of parameter values, resulting in a jerky motion like a socket wrench as I turn the knob several times.

I dug around in the code and found [this line](https://github.com/raphaelquast/beatstep/blob/e9dc0fae942d05b2e8729fbcead3f6be17510a8b/BeatStep_Q.py#L90) which I understand will set the encoder acceleration, presumably just as can be done in the Arturia Midi Control Center. I changed the acceleration value to 2 (which I understand from the comments to be the maximum, "Fast" acceleration) and I did notice a significant reduction in the number of turns required to cover the full range of macro values as compared to 0 acceleration, but I still found this far too slow. It was impossible to traverse the range of values available in my macros in Ableton with anything less than 3-4 turns of an encoder.

I seem to recall seeing somewhere that the relative values sent do not have to be 63 / 65 but can be a greater absolute difference from 64, and I believe this may cause Ableton to traverse parameter values more quickly. Is it feasible to set these greater relative offsets in the Beatstep_Q code and do you believe this would have the desired effect?
